docs/proposals/image-promotion.md

```diff
@@ -0,0 +1,201 @@
+# Image Promotion
+
+## Abstract
+
+This proposal describes the best practices and defines patterns for promoting Docker
+images between different stages (eg. from *development* to *production*).
+This proposal also describes a way how to configure OpenShift to serve as
+a Continuous Delivery tool.
+
+## Constraints and Assumptions
+
+This document describes the following ways to promote images:
+
+* **Based on human intervention**
+  * using the `oc` and `docker` commands manually
+* **Automated by the OpenShift platform**
+  * using enhanced *DeploymentConfig* [lifecycle hooks](https://docs.openshift.org/latest/dev_guide/deployments.html#lifecycle-hooks)
+  * using [ImageChangeTrigger](https://docs.openshift.org/latest/dev_guide/deployments.html#image-change-trigger)
+  * using [Projects](https://docs.openshift.org/latest/dev_guide/projects.html) for different "stages"
+* **Based on external Continuous Delivery tool**
+  * Jenkins
+  * [go.cd](http://go.cd)
+  * [snap](http://snap-ci.com)
+
+Docker images can be promoted (and the use-cases are valid for):
+
+* **Within a single Project**
+* **Between multiple Projects**
+* **Between multiple Clusters**
+
+The Docker image promotion can be done using the Docker [image tags](https://docs.docker.com/userguide/dockerimages/#setting-tags-on-an-image) or [labels](https://docs.docker.com/userguide/labels-custom-metadata/).
+OpenShift also provides more features to add metadata, such as annotations or labels.
```

Comment:
I thought annotations where meant to be used internally by OpenShift or k8s, whereas labels are for users. Unless you're talking about different annotations, in which case we should make it more clear which one you're talking about.